core/src/main/java/org/infinispan/marshall/core/ExternalJBossMarshaller.java

```diff
@@ -0,0 +1,165 @@
+package org.infinispan.marshall.core;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.io.OutputStream;
+
+import org.infinispan.commons.io.ByteBuffer;
+import org.infinispan.commons.marshall.BufferSizePredictor;
+import org.infinispan.commons.marshall.MarshallableTypeHints;
+import org.infinispan.commons.marshall.StreamingMarshaller;
+import org.infinispan.commons.marshall.jboss.ExtendedRiverUnmarshaller;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.jboss.marshalling.ByteInput;
+import org.jboss.marshalling.ByteOutput;
+
+final class ExternalJBossMarshaller implements StreamingMarshaller {
```

Comment:
Not in the scope of this. In fact, external marshaller as concept is likely to go away and be replaced by a user marshaller (see marshalling doc).